pkg/syncer/syncer.go

```diff
@@ -299,6 +324,55 @@ func (c *Controller) handleErr(err error, i interface{}) {
 	klog.Errorf("Dropping key %q after failed retries: %v", i, err)
 }
 
+const nsDelimiter = "--"
+const nsTemplate = "kcp" + nsDelimiter + "%s" + nsDelimiter + "%s"
```

Comment:
Yeah I don't think it would be that hard to store the data in an annotation and write something that would not overflow the namespace name length limit. I don't really think the slightly higher coding cost is worth having a system where if you have a long enough cluster/namespace name, something panics somewhere or the system falls over.

Reply:
ok cool, I'll do that.